File: c/darts/src/darts.c
#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t pos){

    float dist = sqrt((pos.x * pos.x) + (pos.y * pos.y));
    
    if(dist > 10) // fuera del circulo mas grande
        return 0;
    else if(dist > 5)
        return 1;
    else if(dist > 1)
        return 5;
    else    
        return 10;
}

File: java/darts/src/main/java/Darts.java
import java.lang.Math;

class Darts {

    private double x;
    private double y;

    Darts(double x, double y) {
        this.x = x;
        this.y = y;
    }

    int score() {
        double dist = Math.sqrt(x*x + y*y);
        if(dist>10)
            return 0;
        else if(dist>5)
            return 1;
        else if(dist>1)
            return 5;
        else
            return 10;
    }

}
